File: server/generate_cities_from_minshu.py
# -*- coding: utf-8 -*-
"""
从敏舒官方城市列表生成 cities.json
数据来源: PollenForecast/figma/日志.md
"""

import json
import os
import re

# 文件路径
script_dir = os.path.dirname(os.path.abspath(__file__))
minshu_file = os.path.join(script_dir, '..', 'PollenForecast', 'figma', '日志.md')
output_file = os.path.join(script_dir, 'cities.json')

# 读取敏舒官方城市列表
with open(minshu_file, 'r', encoding='utf-8') as f:
    content = f.read()

cities = []
lines = content.strip().split('\n')

# 跳过标题行（第一行是"城市编码\t名称"）
for i, line in enumerate(lines):
    if i == 0:  # 跳过标题行
        continue
    
    line = line.strip()
    if not line:
        continue
    
    # 解析每行：adcode\tname
    parts = line.split('\t')
    if len(parts) >= 2:
        adcode = parts[0].strip()
        name = parts[1].strip()
        
        # 验证 adcode 格式（6位数字）
        if re.match(r'^\d{6}$', adcode) and name:
            cities.append({
                'adcode': adcode,
                'name': name
            })

# 保存为 JSON
with open(output_file, 'w', encoding='utf-8') as f:
    json.dump(cities, f, ensure_ascii=False, indent=2)

print('=' * 40)
print('Done! Generated cities.json from Minshu official list')
print(f'   Source: {minshu_file}')
print(f'   Total cities: {len(cities)}')
print('=' * 40)

File: server/generate_cities_from_minshu.js
/**
 * 从敏舒官方城市列表生成 cities.json
 * 数据来源: PollenForecast/figma/日志.md
 */

const fs = require('fs');
const path = require('path');

// 读取敏舒官方城市列表
const minshuFile = path.join(__dirname, '..', 'PollenForecast', 'figma', '日志.md');
const content = fs.readFileSync(minshuFile, 'utf8');

const cities = [];
const lines = content.split('\n');

// 跳过标题行（第一行是"城市编码	名称"）
for (let i = 1; i < lines.length; i++) {
  const line = lines[i].trim();
  if (!line) continue;
  
  // 解析每行：adcode\tname
  const parts = line.split('\t');
  if (parts.length >= 2) {
    const adcode = parts[0].trim();
    const name = parts[1].trim();
    
    // 验证 adcode 格式（6位数字）
    if (/^\d{6}$/.test(adcode) && name) {
      cities.push({
        adcode: adcode,
        name: name
      });
    }
  }
}

// 保存为 JSON
const outputPath = path.join(__dirname, 'cities.json');
fs.writeFileSync(outputPath, JSON.stringify(cities, null, 2), 'utf8');

console.log('========================================');
console.log('✅ 已从敏舒官方列表生成 cities.json');
console.log(`   数据来源: ${minshuFile}`);
console.log(`   城市数量: ${cities.length} 个`);
console.log('========================================');
console.log('前10个城市:');
cities.slice(0, 10).forEach(city => {
  console.log(`  ${city.adcode}: ${city.name}`);
});
console.log('...');
console.log('后5个城市:');
cities.slice(-5).forEach(city => {
  console.log(`  ${city.adcode}: ${city.name}`);
});
